cirq/contrib/routing/device.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from typing import Iterable, Tuple
+
+import cirq
+import networkx as nx
+
+
+def xmon_device_to_graph(device: cirq.google.XmonDevice) -> nx.Graph:
+    """Gets the graph of an XmonDevice."""
+    return gridqubits_to_graph_device(device.qubits)
+
+
+def get_linear_device_graph(n_qubits: int) -> nx.Graph:
+    """Gets the graph of a linearly connected device."""
+    qubits = cirq.LineQubit.range(n_qubits)
+    edges = [tuple(qubits[i:i + 2]) for i in range(n_qubits - 1)]
+    return nx.Graph(edges)
+
+
+def get_grid_device_graph(width: int, height: int, dx: int = 0,
+                          dy: int = 0) -> nx.Graph:
+    """Gets the graph of a grid of qubits."""
+    return gridqubits_to_graph_device(
+        cirq.GridQubit.rect(width, height, top=dx, left=dy))
```

Comment:
Because I was thinking in Cartesian coordinates rather than matrix indexing. Changed to just delegate all arguments to `GridQubit.rect` for automatic consistency.